collector/config/config.go

```diff
@@ -3,7 +3,20 @@
 
 package config // import "go.opentelemetry.io/ebpf-profiler/collector/config"
 
-import "time"
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"time"
+
+	"go.opentelemetry.io/collector/confmap/xconfmap"
```

Comment:
It seems like with the import of this dependency we increase the binary size from 30108535 bytes (for commit e413d8c4e8b8851ed3addd73f89b3ccd53526f19 on main) to 30929253 bytes (for commit f6f355b9fba94ed397da10747a6504059f69ad14 on this branch). Resulting in an increase of ~0.8 MB.

As xconfmap is used only for the compile time check, can we work around this in some other way?

Reply:
I have removed this type assertion as this is also tested by [this unit test](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/897/files#diff-068b3cac88253595a301fb8238a5d0503bada1ebd38b476868f364d204c7a5c2).